src/Sylius/Bundle/CoreBundle/Fixture/Factory/PromotionExampleFactory.php

```diff
@@ -118,20 +144,20 @@ protected function configureOptions(OptionsResolver $resolver): void
             ->setAllowedTypes('channels', 'array')
             ->setNormalizer('channels', LazyOption::findBy($this->channelRepository, 'code'))
             ->setDefined('rules')
-            ->setNormalizer('rules', function (Options $options, array $rules): array {
-                if (empty($rules)) {
-                    return [[]];
+            ->setNormalizer('rules', function (Options $options, ?array $rules): array {
+                if (null === $rules) {
+                    return [];
                 }
 
-                return $rules;
+                return empty($rules) ? [[]] : $rules;
```

Comment:
I can see that it's been like this before, but is there a particular reason why it's not a simple array?

Reply:
Ok, it has been working in a way that if I pass an empty array, then this `normalizer` returns an empty array within an array, which previously created a promotion with at least one action/rule. For the purpose of using this in Behat tests, where I need to create a promotion without any actions/rules, I've added the option to pass a null value. In this case, it will return an empty array, so no actions/rules are added. If someone passes an empty array, the behavior will remain as it was before.